### PR TITLE
(feat) O3-4249: Add component to support rendering an icon if it exists

### PR DESCRIPTION
**Pull request by ibacher**
# Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. Ensure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing#contributing-guidelines) label (such as `feat`, `fix`, or `chore`, among others). See existing PR titles for inspiration.

## For changes to apps

- [x] My work conforms to the [**O3 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://om.rs/o3ui).

## If applicable

- [ ] My work includes tests or is validated by existing tests.
- [ ] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary

This is pretty straight-forward. It adds a couple of utility components:

* `RenderIfValueIsTruthy` — Does what it says; either renders the provided children if the `value` prop is truthy or else renders a specified fallback (default nothing). This is just a building block for the functionality added here, but it seemed like it may be more broadly useful.
* `MaybeIcon` — This is basically a combination of `RenderIfValueIsTruthy` and the existing `Icon` component, so that it renders as an icon if the icon specified as a string is one of the icons provided and as nothing if the icon isn't defined.
* `MaybePictogram` — Same thing, but wrapping pictograms.

The intended use case is to support configuration-driven icons, for example:

```tsx
const MyComponentWithIcon() {
  const config = useConfig();

  return (
    <>
      <MaybeIcon icon={`omrs-icon-${config.icon}`} size=20 />
      {t('label', 'My Action')}
    </>
  );
}
```

As long as `config.icon` resolves to a known icon id, e.g., `arrow-up`, this will render exactly like the `ArrowUpIcon` component, but if `config.icon` specifies something that doesn't exist, nothing will be rendered at all.

## Other

This also adds some docs for `UserHasAccess` which is kind of similar to `RenderIfValueIsTruthy`.

### CODE REVIEW

**Review by vasharma05 — APPROVED** [review]
Thank you @ibacher for working on this. I was about to discuss about the same in a few days, so happy to see this!

File: packages/framework/esm-styleguide/src/icons/icons.tsx [hunk 1]
```diff
@@ -698,6 +699,48 @@ export const RadiologyIcon = ImageMedicalIcon;
...
+      const container = document.getElementById('omrs-svgs-container');
+      const callback: MutationCallback = (mutationList) => {
+        for (const mutation of mutationList) {
+          if (mutation.type === 'childList') {
+            iconRef.current = document.getElementById(icon);
+          }
+        }
+      };
+
+      const observer = new MutationObserver(callback);
```
Comment (ibacher) [inline packages/framework/esm-styleguide/src/icons/icons.tsx:728]: Note to self: I'm not sure about the performance implications of having a per-icon mutation observer vs, say, a global mutation observer that could have callbacks registered with it. This is, however, easier to write.

File: packages/framework/esm-styleguide/src/icons/icons.tsx [hunk 2]
```diff
@@ -698,6 +699,48 @@ export const RadiologyIcon = ImageMedicalIcon;
  */
 export const ShoppingCartAddItemIcon = ShoppingCartArrowDownIcon;
 
+/**
+ * This is a utility component that takes an `icon` and render it if the sprite for the icon
```
Comment (denniskigen) [inline packages/framework/esm-styleguide/src/icons/icons.tsx:703]:
```suggestion
 * This is a utility component that takes an `icon` and renders it if the sprite for the icon
```

nit: typo

File: packages/framework/esm-styleguide/src/pictograms/pictograms.tsx [hunk 3]
```diff
@@ -80,3 +56,71 @@ export const ServiceQueuesPictogram = memo(
     return <Pictogram ref={ref} pictogram="omrs-pict-service-queues" pictogramProps={props} />;
   }),
 );
+
+/**
+ * This is a utility component that takes an `icon` and render it if the sprite for the icon
+ * is available. The goal is to make it easier to conditionally render configuration-specified icons.
```
Comment (denniskigen) [inline packages/framework/esm-styleguide/src/pictograms/pictograms.tsx:62]:
```suggestion
 * This is a utility component that takes an `pictogram` and render it if the sprite for the pictogram
 * is available. The goal is to make it easier to conditionally render configuration-specified pictograms.
```

File: packages/framework/esm-styleguide/src/pictograms/pictograms.tsx [hunk 4]
```diff
@@ -80,3 +56,71 @@ export const ServiceQueuesPictogram = memo(
...
+ *
+ * @example
+ * ```tsx
+ *   <MaybeIcon icon='omrs-icon-baby' className={styles.myIconStyles} />
```
Comment (denniskigen) [inline packages/framework/esm-styleguide/src/pictograms/pictograms.tsx:66]:
```suggestion
 *   <MaybePictogram pictogram='omrs-icon-baby' className={styles.myPictogramStyles} />
```

File: packages/framework/esm-styleguide/src/icons/icons.tsx [hunk 5]
```diff
@@ -698,6 +699,48 @@ export const RadiologyIcon = ImageMedicalIcon;
...
+ * is available. The goal is to make it easier to conditionally render configuration-specified icons.
+ *
+ * @example
+ * ```tsx
+ *   <MaybeIcon icon='omrs-icon-baby' className={styles.myIconStyles} />
+ * ```
+ */
+export const MaybeIcon = memo(
```
Comment (denniskigen) [inline packages/framework/esm-styleguide/src/icons/icons.tsx:711]:
```suggestion
export const MaybeIcon = memo(
```

nit: Maybe `LoadableIcon` reads better here?

File: packages/framework/esm-styleguide/src/icons/icons.tsx (hunk 1 above)
Comment (denniskigen) [reply packages/framework/esm-styleguide/src/icons/icons.tsx:728]: Guess we'll find out!

**Review by denniskigen — APPROVED** [review]
Thanks, @ibacher. LGTM. I've left a few persnickety comments.

**Review by NethmiRodrigo — APPROVED** [review]

**Pull request merged**